### DISCUSSION TITLE
Native sdk versions

### DISCUSSION THREAD

**Discussion by appsgenie** [thebergamo/react-native-fbsdk-next#330] [Q&A]
How do you tell what version of the facebook android sdk react-native-fbsdk-next is using? Basically looking to know the native underlying versions for each platform..

**Comment by mikehardy** [comment]
You can pin it if you want in your gradle files with gradle variable `facebookSdkVersion`, we'll defer to a user-provided variable if provided (though you'll be responsible for moving it yourself at that point, on updates): https://github.com/thebergamo/react-native-fbsdk-next/blob/3ec1bd3d4512f1cf4aceb23c59724f4d1d17d4c5/android/build.gradle#L47

Otherwise you can see we let the dependency "slip" on minor versions with a plus for semver minor, so it's variable depending on when you do your build. That violates reproducible build policies if those are important to you but it's a tradeoff as pinning the whole version here means you have to wait here for bugfixes to come in, we've chosen fewer releases here and letting minors / patches flow through to consumers.

To directly answer your question, it's a general gradle feature for android apps, assuming you have the default `android/app` as your native android directories then you want to run `./gradlew :app:dependencies` to get your dependencies report showing what everything resolves to.

I find the output of that command to be gigantic in practice, so `./gradlew :app:dependencies > dependencies.txt` and then you can browse through it.

**Reply by appsgenie** [reply]
Got it. So in simple terms, if I just install the package and don't mess with any versioning and what to the the default, it will use the value that is listed on that `safeExtGet` line?
Not sure if this matters, but I am using it with Expo's EAS system.

**Reply by mikehardy** [reply]
that is correct, you will get the most recent release in their v15 major version release series at time of build

**Reply by appsgenie** [reply]
Do you happen to know if this is an issue with this react native package or with the native SDK? I am trying to debug it and understand in what version it is happening but not sure where to start... 
From a crash report when trying to login:
```
Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent. Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the Pendinglntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
```

**Reply by mikehardy** [reply]
With apologies, this is not really an open tech support thread, I'm a developer with deadlines as well just trying to help

Search first: https://github.com/thebergamo/react-native-fbsdk-next/issues?q=IMMUTABLE

It's fixed in current versions, for this module

**Reply by appsgenie** [reply]
understood thank you for your time on this, yes I've seen one of those issues and it got me into looking at what version I am using and that's how I landed here. apologies. I think this would have been better suited for the Issues section (but of course after more research into existing open issues). :pray:

**Reply by appsgenie** [reply]
sorry to bother again, but where would I address issues with the current version that might have the fix for this? Like after upgrading, it fails building android at the gradle step :(

**Reply by mikehardy** [reply]
You would check stackoverflow for the exact build error (which you haven't posted here, so I can't even offer help at a glance) and/or compare with a "known good" build, that is an android build that works on current versions which we have right here in this repo if you clone it out and run it.

Stated differently: this module does not have issues with the current versions, but your project might and I'd look at a project that worked first then compare/contrast with whatever you have set up in your project.

**Reply by appsgenie** [reply]
Yes I can definitely admit it's beyond my knowledge in researching this so trying stackoverflow as well. I am using Expo and I think I saw your comment in another issue that some people are having issues with latest version vs Expo. But I can't yet tell if it's related to that. All I did is build it with `"react-native-fbsdk-next": "^10.1.0"` and it succeeded and then with `"react-native-fbsdk-next": "^11.0.0"` and it failed with the errors below. So maybe some new dependency or some gradle version mismatch? 

```
[stderr] Execution failed for task ':app:checkReleaseDuplicateClasses'.
[stderr] > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
[stderr]    > Duplicate class ....
```
then some
```
[stderr] BUILD FAILED in 9m 3s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
```

If any if this rings any bells for you would appreciate some direction :pray:
